### ISSUE TITLE
Work out URL structure

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#28939]

**Comment by courier-new** [comment]
I decided to think about these URLs by exploring the user flows that make use of them. Rather than hold myself to the constraints of our existing URL scheme and high-level page layout, I decided to aim for something a bit more radical and approached this closer to the angle of starting from scratch. This helped me break down some of the issues we've run into with multiple nested levels of tabs and somewhat redundant/confusing information being all up on the screen at once.

I ended up putting together a [prototype](https://ssbc-proto.vercel.app/courier-new/batch-changes) to demonstrate what I came up with. Though it doesn't capture _every_ piece of information that might be surfaced for a batch change, it captures the biggest pieces, I think. 

IMO it's easier to grasp this by playing with the prototype, but here's my revision on Erik's take above:

- `/batch-changes`: Global list of batch changes. This includes draft ones, but only the ones visible to the viewer.
- `/batch-changes/create`: Global create page. This is a page that is easy to link to, and the namespace is selectable in the create form.
- `/{users|orgs}/:ownerName/batch-changes`: List of batch changes owned by the entity with ownerName. This includes draft ones, but only the ones visible to the viewer.
- `/{users|orgs}/:ownerName/batch-changes/:name`: The detail page of the batch change with name name in namespace ownerName -- defaults to changesets view, but will redirect to another URL if the batch change has not had a spec applied yet 
- `/{users|orgs}/:ownerName/batch-changes/create`: Create page of the namespace. Only accessible by people that can create a batch change in that namespace. The namespace can not be changed in that version of the create form.
- `/{users|orgs}/:ownerName/batch-changes/:name/edit`: The detail page with the SSBC editor (step 1) open, to create a new batch spec.
- `/{users|orgs}/:ownerName/batch-changes/:name/close`: As before.
- `/{users|orgs}/:ownerName/batch-changes/:name/executions`: The details page with the batch spec executions list open.
- `/{users|orgs}/:ownerName/batch-changes/:name/executions/:executionID`: The details page with the execution screen (step 2) open for the specified batch spec execution.
- `/{users|orgs}/:ownerName/batch-changes/:name/executions/:executionID/spec`: The details page with the SSBC editor (step 1) open in read-only mode for the specified batch spec execution. 
- `/{users|orgs}/:ownerName/batch-changes/:name/executions/:executionID/preview`: The details page with the preview (step 3) open.
- `/{users|orgs}/:ownerName/batch-changes/:name/burndown`: The details page with the burndown chart displayed.
- `/{users|orgs}/:ownerName/batch-changes/:name/settings`: The details page with the batch spec settings open.
- `/{users|orgs}/:ownerName/batch-changes/:name/bulk-actions`: The details page with the bulk actions list displayed.

**Comment by eseliger** [comment]
Nice, like that proposal.

> /{users|orgs}/:ownerName/batch-changes/:name/executions/:executionID/spec: The details page with the SSBC editor (step 1) open in read-only mode for the specified batch spec execution.

I wonder if there also needs to be a route that allows to edit the existing spec somehow 🤔 

Other than that, I think this is great and we can get started implementing this as a next step. Who should own that?

**Comment by courier-new** [comment]
> I wonder if there also needs to be a route that allows to edit the existing spec somehow

I suppose there's no reason you couldn't click an "Edit" button from that spec view that opens `/{users|orgs}/:ownerName/batch-changes/:name/edit` pre-filled with the batch spec YAML from the existing spec you just came from!

I'm cool with owning this unless you're super excited about doing it yourself! 🙂 I have some useful work for the page layout part of this stashed on a branch from before the holidays too, so with any hope, that's still relevant and gives me a bit of a head start.

**Comment by eseliger** [comment]
Yeah no that makes sense to me, feel free to take it :) Was just gonna make sure.
